term_unix_ioctl.go

```diff
@@ -0,0 +1,58 @@
+// +build linux darwin freebsd openbsd netbsd dragonfly
+
+// Copyright 2013-2015 Bowery, Inc.
+// Copyright 2017 Attila Fülöp <attila@fueloep.org>
+
+package prompt
+
+import (
+	"os"
+	"syscall"
+	"unsafe"
+	"errors"
+)
+
+type Termios syscall.Termios
+
+// setTermios does the system dependent ioctl calls
+func getTermios(fd uintptr, req uintptr) (*Termios, error) {
+	termios := new(syscall.Termios)
+	_, _, err := syscall.Syscall(syscall.SYS_IOCTL, fd, req,
+		uintptr(unsafe.Pointer(termios)))
+
+	if err != 0 {
+		return nil, errors.New(err.Error())
```

Comment:
Yes, @gfrey,  you are right.  Errno implements the error interface, so we can simply return err. We already made sure not to return a zero Errno here. Sorry for missing that.